### DISCUSSION TITLE
Feature Request: `list_aggregate` takes lambda functions

### DISCUSSION THREAD

**Discussion by groceryheist** [duckdb/duckdb#6797] [Ideas]
Currently `list_aggregate` takes a second function name as an argument.  But some aggregation functions take more than one argument. For example, `string_agg` takes a separator that defaults to ','.  If `list_aggregate` could take a lambda that resolves to an aggregator then it could support other separators.

**Comment by Mytherin** [comment]
You can already pass additional parameters to `list_aggregate`, although they have to be constants:

```sql
D select list_aggregate([1,2,3], 'string_agg', '|') as l;
┌─────────┐
│    l    │
│ varchar │
├─────────┤
│ 1|2|3   │
└─────────┘
```

Is that sufficient, or could you perhaps expand on what you are requesting?

**Comment by groceryheist** [comment]
Oh! That will work for my current use-case.  I guess this is more of a documentation gap then.
